mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/building/impl/FieldModelContributorIndirectContextImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.mapping.building.impl;
+
+import org.hibernate.search.engine.backend.types.dsl.ScaledNumberIndexFieldTypeContext;
+import org.hibernate.search.engine.backend.types.dsl.StandardIndexFieldTypeContext;
+import org.hibernate.search.mapper.pojo.bridge.ValueBridge;
+import org.hibernate.search.mapper.pojo.mapping.building.spi.FieldModelContributorIndirectContext;
+
+/**
+ * Provides an <b>indirect</b> way to contribute on {@link StandardIndexFieldTypeContext}.
+ *
+ * Useful for the case when the contribution is suitable of translation.
+ * For instance, if the value must be converted using a {@link ValueBridge}.
+ *
+ * Moreover, it could be useful whether the applying of the contribution
+ * could depends on other contributions applied previously
+ * on the same {@link StandardIndexFieldTypeContext}.
+ * For instance, a default value should not be applied
+ * if the same value has been provided explicitly.
+ *
+ * @param <F> The type of raw index field values, on the index side of the bridge.
+ */
+public class FieldModelContributorIndirectContextImpl<F> implements FieldModelContributorIndirectContext {
+
+	private final ValueBridge<?, F> bridge;
+	private final StandardIndexFieldTypeContext<?, ? super F> fieldTypeContext;
+
+	private boolean useDefaultDecimalScale = true;
+
+	public FieldModelContributorIndirectContextImpl(ValueBridge<?, F> bridge, StandardIndexFieldTypeContext<?, ? super F> fieldTypeContext) {
+		this.bridge = bridge;
+		this.fieldTypeContext = fieldTypeContext;
+	}
+
+	@Override
+	public void indexNullAs(String value) {
+		fieldTypeContext.indexNullAs( bridge.parse( value ) );
+	}
+
+	@Override
+	public void decimalScale(int decimalScale) {
+		if ( fieldTypeContext instanceof ScaledNumberIndexFieldTypeContext ) {
+			( (ScaledNumberIndexFieldTypeContext) fieldTypeContext ).decimalScale( decimalScale );
+			useDefaultDecimalScale = false;
+		}
+	}
+
+	@Override
+	public void defaultDecimalScale(int decimalScale) {
+		if ( useDefaultDecimalScale && fieldTypeContext instanceof ScaledNumberIndexFieldTypeContext ) {
```

Comment:
It's a nice a solution, but I wonder if it's better than the obvious one of applying contributions in the right order so that the last one wins.

We have three sources of contributions:

* the field bridge itself, which may apply a "hard-coded" decimal scale if it's a custom bridge.
* the `@ScaledNumberField` annotation (or programmatic mapping equivalent)
* the `@Column` annotation

I think what we want is this precedence:

1. `@ScaledNumberField` (takes precedence over everything)
2. Field bridge
3. `@Column` (only applied as a last-resort default)

What you implemented is this:

1. `@ScaledNumberField` (takes precedence over everything)
2. `@Column`
3. Field bridge (only applied as a last-resort default)

The thing is, if we simply applied the default before applying the `@ScaledNumberField`, we would get exactly the same precedence, with a much simpler implementation:

* Apply bridge
* Call `.decimalScale( <value from @Column> )` if a value was provided
* Call `.decimalScale( <value from @ScaledNumberField> )` if a value was provided

Now, if we wanted the Field bridge to take precedence over the default, then we would need to communicate the defaults to the backend somehow. This is doable: we would basically only need to pass an object with a `getDefaultDecimalScale()` method to `org.hibernate.search.engine.mapper.mapping.building.spi.IndexBindingContext#getTypeFactory`. But it would add a bit of complexity. Do we want to go there? Let's discuss it, maybe?